2-Draft-Accepted/RFCxxxx-PowerShellGet-3.0.md

```diff
@@ -0,0 +1,242 @@
+---
+RFC: RFCxxxx
+Author: Steve Lee
+Status: Draft
+SupercededBy: N/A
+Version: 0.1
+Area: PowerShellGet
+Comments Due: 7/30
+Plan to implement: Yes
+---
+
+# PowerShellGet 3.0 Module
+
+PowerShellGet has enabled an active community to publish and install PowerShell
+Modules.
+However, it has some architectural and design limitations that prevent it from
+moving forward more quickly.
+This RFC proposes some significant changes to address this.
+
+> [!NOTE]
+> There are no intentions to break existing PSGet v2 users so existing
+> users can stay on PSGet v2 and that will continue to work.
+
+## Motivation
+
+    As a PowerShell user,
+    I can discover how to install missing cmdlets,
+    so that I can be more productive.
+
+    As a PowerShellGet contributor,
+    I can more easily contribute code,
+    so that PowerShellGet evolves more quickly.
+
+## User Experience
+
+```powershell
+PS> Get-Satisfaction
+Get-Satisfaction : The term 'Get-Satisfaction' is not recognized as the name of a cmdlet, function, script file, or operable program.
+Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
+At line:1 char:1
++ Get-Satisfaction
++ ~~~
++ CategoryInfo          : ObjectNotFound: (Get-Satisfaction:String) [], CommandNotFoundException
++ FullyQualifiedErrorId : CommandNotFoundException
+
+Suggestion: You can obtain `Get-Satisfaction` by running `Install-PSResource HappyDays`.
+
+PS> Install-PSResource HappyDays
+```
+
+## Specification
+
+### Rewrite of PowerShellGet
+
+PowerShellGet is currently written as PowerShell script with a dependency on PackageManagement.
+Proposal is to write PSResource in C# to reduce complexity and make easier to maintain.
+In addition, remove dependency on PackageManagement completely as well as dependency on
+nuget.exe.
+This module will only use REST APIs to get and publish nupkgs.
+This module would be shipped in PowerShell 7.
+
+### Side-by-side with PowerShellGet
+
+To ease transition due to the enormity of the breaking change, this module will
```

Comment:
I'm a bit lost what the actual breaking change is from a user perspective. Could you elaborate?

Reply:
The cmdlets will be different (even though I've agreed that `Install-Module` and `Find-Module` will be retained).  When I refresh this doc based on the feedback, I'll soften this as the breaking changes will be less now.  For example, I was originally following the `apt` model of requiring manual updates of the cache, but have since decided auto-update is a better experience.  This line wasn't updated to reflect any of this.

Reply:
@SteveL-MSFT are you sure you want to do that? 

Are we willing to implement _all_ the parameters of the old commands in the new module without breaking anything, or do you think there's a subset that will keep "most" of the use cases, without causing really awful error messages when people try to use the missing parts?

I'm leaning toward this new module just being a new module, and let people install the old one (which has 100% compatibility) if they are not ready to update their code.

Reply:
@Jaykul I haven't gone through each and every parameter, so I would like to drop ones that no longer made sense or provided a bad user experience.  My current thinking is to still call this PowerShellGet 3.0 instead of something else and you can use it side-by-side with v2.